var/spack/repos/builtin/packages/libxml2/package.py

```diff
@@ -27,7 +27,7 @@ class Libxml2(AutotoolsPackage):
     depends_on('zlib')
     depends_on('xz')
 
-    depends_on('python+shared', when='+python')
+    depends_on('python+shared~libxml2', when='+python')
```

Comment:
I would also add a comment/reference before this line warning about the cyclic dependency